### ISSUE TITLE
Fix PlayerAuthInput Packet

### ISSUE THREAD

**Issue by theaddonn** [bedrock-crustaceans/bedrockrs#33]
The packet does mostly work, but other cases such as when inventory transactions or block actions should be read additionally are not properly handled

**Comment by ismaileke** [comment]
This is irrelevant, but
```php
PocketMine->BedrockProtocol

$out->putUnsignedVarInt($this->interactionMode); <----
if($this->playMode === PlayMode::VR){
    assert($this->vrGazeDirection !== null);
    $out->putVector3($this->vrGazeDirection); <----
}
```

```rust
let play_mode = match VAR::<u32>::proto_deserialize(stream)?.into_inner() {
0 => PlayMode::Normal,
            1 => PlayMode::Teaser,
            2 => PlayMode::Screen,
            3 => PlayMode::Viewer,
            4 => {
                let vr_gaze_direction = ProtoCodec::proto_deserialize(stream)?; <-------------------------------

                PlayMode::Reality(vr_gaze_direction)
            }
            5 => PlayMode::Placement,
            6 => PlayMode::LivingRoom,
            7 => PlayMode::ExitLevel,
            8 => PlayMode::ExitLevelLivingRoom,
            other => {
                return Err(ProtoCodecError::InvalidEnumID(
                    other.to_string(),
                    String::from("PlayMode"),
                ))
            }
        };
        let interaction_model = InteractionModel::proto_deserialize(stream)?; <--------------
```
Shouldn't you put `vr_gaze_direction` after the `interaction_model` variable?

If I'm wrong, it's probably due to my rust knowledge. 🛠️

**Comment by theaddonn** [comment]
You are completely correct!
Thanks for pointing it out
![image](https://github.com/user-attachments/assets/d74bbdef-5ffe-4823-90ff-3c59b0e93b23)

**Comment by theaddonn** [comment]
if would be really helpful if someone could do a pr for this!

**Comment by theaddonn** [comment]
One fix has been implemented https://github.com/bedrock-crustaceans/bedrockrs/pull/47
> but other cases such as when inventory transactions or block actions should be read additionally are not properly handled
This is still an issue though

**Comment by RadiatedMonkey** [comment]
> One fix has been implemented #47
> 
> > but other cases such as when inventory transactions or block actions should be read additionally are not properly handled
> 
> This is still an issue though

What is the exact issue here? I should have a fully working PlayerAuthInput packet in my software so I could take a look at this and see if I can fix it

**Comment by theaddonn** [comment]
Well, Block actions and Inventoty actions are not handled at all right now. They should be variants with extra structs that store the data in the InputData enum...
